### PR TITLE
修复windows下vscode调试go程序名无后缀的情况导致程序crash

### PR DESCRIPTION
**Pull request by kevin2027**

### CODE REVIEW

**Comment by yaofei517** [comment]
调试通过

File: src/os/src/windows/wSemphone.c [hunk 1]
```diff
@@ -50,7 +50,10 @@ int32_t taosGetCurrentAPPName(char *name, int32_t* len) {
   char filepath[1024] = {0};
 
   GetModuleFileName(NULL, filepath, MAX_PATH);
-  *strrchr(filepath,'.') = '\0';
+  char* sub=strrchr(filepath,'.');
+  if(sub!=NULL){
```
Comment (zitsen) [inline src/os/src/windows/wSemphone.c]: Thanks, please use consist format style with other codes.

```c
char* sub = strchr(filepath, '.');
if (sub != null) {
  *sub  = '\0';
}
```

**Comment by sangshuduo** [comment]
wSemphone.c

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared\\ws2def.h(103): error C2220: warning treated as error - no 'object' file generated

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared\\ws2def.h(103): warning C4005: 'AF\_IPX': macro redefinition

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(457): note: see previous definition of 'AF\_IPX'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared\\ws2def.h(147): warning C4005: 'AF\_MAX': macro redefinition

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(476): note: see previous definition of 'AF\_MAX'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared\\ws2def.h(185): warning C4005: 'SO\_DONTLINGER': macro redefinition

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(399): note: see previous definition of 'SO\_DONTLINGER'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared\\ws2def.h(238): error C2011: 'sockaddr': 'struct' type redefinition

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(482): note: see declaration of 'sockaddr'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared\\ws2def.h(440): error C2059: syntax error: 'constant'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared\\ws2def.h(575): warning C4005: 'IN\_CLASSA': macro redefinition

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(284): note: see previous definition of 'IN\_CLASSA'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared\\ws2def.h(581): warning C4005: 'IN\_CLASSB': macro redefinition

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(290): note: see previous definition of 'IN\_CLASSB'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared\\ws2def.h(587): warning C4005: 'IN\_CLASSC': macro redefinition

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(296): note: see previous definition of 'IN\_CLASSC'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared\\ws2def.h(598): warning C4005: 'INADDR\_ANY': macro redefinition

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(301): note: see previous definition of 'INADDR\_ANY'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared\\ws2def.h(600): warning C4005: 'INADDR\_BROADCAST': macro redefinition

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(303): note: see previous definition of 'INADDR\_BROADCAST'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared\\ws2def.h(636): error C2011: 'sockaddr\_in': 'struct' type redefinition

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(309): note: see declaration of 'sockaddr\_in'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(136): error C2011: 'fd\_set': 'struct' type redefinition

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(65): note: see declaration of 'fd\_set'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(156): warning C4005: 'FD\_CLR': macro redefinition

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(94): note: see previous definition of 'FD\_CLR'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(171): warning C4005: 'FD\_SET': macro redefinition

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(99): note: see previous definition of 'FD\_SET'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(180): error C2011: 'timeval': 'struct' type redefinition

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(108): note: see declaration of 'timeval'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(236): error C2011: 'hostent': 'struct' type redefinition

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(164): note: see declaration of 'hostent'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(249): error C2011: 'netent': 'struct' type redefinition

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(177): note: see declaration of 'netent'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(256): error C2011: 'servent': 'struct' type redefinition

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(184): note: see declaration of 'servent'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(268): error C2011: 'protoent': 'struct' type redefinition

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(196): note: see declaration of 'protoent'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(364): error C2011: 'WSAData': 'struct' type redefinition

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(319): note: see declaration of 'WSAData'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(462): error C2011: 'sockproto': 'struct' type redefinition

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(491): note: see declaration of 'sockproto'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(504): error C2011: 'linger': 'struct' type redefinition

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(528): note: see declaration of 'linger'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(517): warning C4005: 'SOMAXCONN': macro redefinition

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(541): note: see previous definition of 'SOMAXCONN'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(552): warning C4005: 'FD\_READ': macro redefinition

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(559): note: see previous definition of 'FD\_READ'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(555): warning C4005: 'FD\_WRITE': macro redefinition

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(560): note: see previous definition of 'FD\_WRITE'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(558): warning C4005: 'FD\_OOB': macro redefinition

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(561): note: see previous definition of 'FD\_OOB'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(561): warning C4005: 'FD\_ACCEPT': macro redefinition

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(562): note: see previous definition of 'FD\_ACCEPT'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(564): warning C4005: 'FD\_CONNECT': macro redefinition

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(563): note: see previous definition of 'FD\_CONNECT'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(567): warning C4005: 'FD\_CLOSE': macro redefinition

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(564): note: see previous definition of 'FD\_CLOSE'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(1625): error C2375: 'accept': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(739): note: see declaration of 'accept'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(1647): error C2375: 'bind': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(744): note: see declaration of 'bind'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(1668): error C2375: 'closesocket': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(749): note: see declaration of 'closesocket'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(1685): error C2375: 'connect': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(751): note: see declaration of 'connect'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(1706): error C2375: 'ioctlsocket': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(756): note: see declaration of 'ioctlsocket'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(1729): error C2375: 'getpeername': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(761): note: see declaration of 'getpeername'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(1750): error C2375: 'getsockname': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(766): note: see declaration of 'getsockname'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(1771): error C2375: 'getsockopt': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(771): note: see declaration of 'getsockopt'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(1796): error C2375: 'htonl': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(778): note: see declaration of 'htonl'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(1813): error C2375: 'htons': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(780): note: see declaration of 'htons'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(1831): error C2375: 'inet\_addr': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(782): note: see declaration of 'inet\_addr'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(1849): error C2375: 'inet\_ntoa': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(784): note: see declaration of 'inet\_ntoa'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(1949): error C2375: 'listen': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(786): note: see declaration of 'listen'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(1968): error C2375: 'ntohl': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(790): note: see declaration of 'ntohl'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(1985): error C2375: 'ntohs': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(792): note: see declaration of 'ntohs'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(2002): error C2375: 'recv': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(794): note: see declaration of 'recv'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(2025): error C2375: 'recvfrom': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(800): note: see declaration of 'recvfrom'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(2052): error C2375: 'select': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(808): note: see declaration of 'select'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(2077): error C2375: 'send': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(815): note: see declaration of 'send'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(2100): error C2375: 'sendto': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(821): note: see declaration of 'sendto'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(2127): error C2375: 'setsockopt': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(829): note: see declaration of 'setsockopt'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(2152): error C2375: 'shutdown': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(836): note: see declaration of 'shutdown'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(2172): error C2375: 'socket': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(840): note: see declaration of 'socket'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(2197): error C2375: 'gethostbyaddr': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(847): note: see declaration of 'gethostbyaddr'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(2219): error C2375: 'gethostbyname': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(852): note: see declaration of 'gethostbyname'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(2236): error C2375: 'gethostname': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(854): note: see declaration of 'gethostname'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(2276): error C2375: 'getservbyport': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(858): note: see declaration of 'getservbyport'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(2295): error C2375: 'getservbyname': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(862): note: see declaration of 'getservbyname'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(2314): error C2375: 'getprotobynumber': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(866): note: see declaration of 'getprotobynumber'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(2331): error C2375: 'getprotobyname': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(868): note: see declaration of 'getprotobyname'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(2351): error C2375: 'WSAStartup': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(872): note: see declaration of 'WSAStartup'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(2371): error C2375: 'WSACleanup': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(876): note: see declaration of 'WSACleanup'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(2388): error C2375: 'WSASetLastError': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(878): note: see declaration of 'WSASetLastError'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(2405): error C2375: 'WSAGetLastError': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(880): note: see declaration of 'WSAGetLastError'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(2426): error C2375: 'WSAIsBlocking': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(882): note: see declaration of 'WSAIsBlocking'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(2444): error C2375: 'WSAUnhookBlockingHook': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(884): note: see declaration of 'WSAUnhookBlockingHook'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(2462): error C2375: 'WSASetBlockingHook': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(886): note: see declaration of 'WSASetBlockingHook'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(2480): error C2375: 'WSACancelBlockingCall': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(888): note: see declaration of 'WSACancelBlockingCall'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(2498): error C2375: 'WSAAsyncGetServByName': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(890): note: see declaration of 'WSAAsyncGetServByName'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(2526): error C2375: 'WSAAsyncGetServByPort': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(898): note: see declaration of 'WSAAsyncGetServByPort'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(2554): error C2375: 'WSAAsyncGetProtoByName': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(906): note: see declaration of 'WSAAsyncGetProtoByName'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(2580): error C2375: 'WSAAsyncGetProtoByNumber': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(913): note: see declaration of 'WSAAsyncGetProtoByNumber'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(2606): error C2375: 'WSAAsyncGetHostByName': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(920): note: see declaration of 'WSAAsyncGetHostByName'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(2632): error C2375: 'WSAAsyncGetHostByAddr': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(927): note: see declaration of 'WSAAsyncGetHostByAddr'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(2662): error C2375: 'WSACancelAsyncRequest': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(936): note: see declaration of 'WSACancelAsyncRequest'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock2.h(2680): error C2375: 'WSAAsyncSelect': redefinition; different linkage

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(938): note: see declaration of 'WSAAsyncSelect'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared\\ws2ipdef.h(135): warning C4005: 'IP\_TOS': macro redefinition

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(349): note: see previous definition of 'IP\_TOS'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared\\ws2ipdef.h(136): warning C4005: 'IP\_TTL': macro redefinition

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(348): note: see previous definition of 'IP\_TTL'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared\\ws2ipdef.h(137): warning C4005: 'IP\_MULTICAST\_IF': macro redefinition

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(343): note: see previous definition of 'IP\_MULTICAST\_IF'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared\\ws2ipdef.h(138): warning C4005: 'IP\_MULTICAST\_TTL': macro redefinition

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(344): note: see previous definition of 'IP\_MULTICAST\_TTL'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared\\ws2ipdef.h(139): warning C4005: 'IP\_MULTICAST\_LOOP': macro redefinition

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(345): note: see previous definition of 'IP\_MULTICAST\_LOOP'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared\\ws2ipdef.h(140): warning C4005: 'IP\_ADD\_MEMBERSHIP': macro redefinition

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(346): note: see previous definition of 'IP\_ADD\_MEMBERSHIP'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared\\ws2ipdef.h(141): warning C4005: 'IP\_DROP\_MEMBERSHIP': macro redefinition

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(347): note: see previous definition of 'IP\_DROP\_MEMBERSHIP'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared\\ws2ipdef.h(142): warning C4005: 'IP\_DONTFRAGMENT': macro redefinition

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(350): note: see previous definition of 'IP\_DONTFRAGMENT'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared\\ws2ipdef.h(738): error C2011: 'ip\_mreq': 'struct' type redefinition

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\\winsock.h(360): note: see declaration of 'ip\_mreq'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared\\ws2ipdef.h(914): error C2061: syntax error: identifier 'IPPROTO'

C:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared\\ws2ipdef.h(918): error C2059: syntax error: '}'

Defining type struct \_stat64i32\_ns

Defining type struct \_stat64\_ns

NMAKE : fatal error U1077: 'C:\\PROGRA~2\\MI0E91~1.0\\VC\\bin\\amd64\\cl.exe' : return code '0x2'

Stop.

NMAKE : fatal error U1077: '"C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\BIN\\amd64\\nmake.exe"' : return code '0x2'

Stop.

NMAKE : fatal error U1077: '"C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\BIN\\amd64\\nmake.exe"' : return code '0x2'

Stop.

Command exited with code 2

**Review by sangshuduo — CHANGES_REQUESTED** [review]
please fix compile error.

**Comment by sangshuduo** [comment]
@yaofei517  thanks for your contribution.

**Comment by yu285** [comment]
您好，我是涛思数据的工作人员，方便留一下联系方式吗？可以拉您进我们的贡献者群，和大家一起交流技术。

**Comment by yaofei517** [comment]
> 您好，我是涛思数据的工作人员，方便留一下联系方式吗？可以拉您进我们的贡献者群，和大家一起交流技术。
好的，微信：huanliu517 请拉我进群

**Pull request merged**